src/main/java/de/tum/in/www1/artemis/domain/lti/Lti13LaunchRequest.java

```diff
@@ -24,8 +25,10 @@ public Lti13LaunchRequest(OidcIdToken ltiIdToken, String clientRegistrationId) {
         this.sub = ltiIdToken.getClaim("sub");
         this.deploymentId = ltiIdToken.getClaim(Claims.LTI_DEPLOYMENT_ID);
 
-        JSONObject resourceLinkClaim = ltiIdToken.getClaim(Claims.RESOURCE_LINK);
-        this.resourceLinkId = resourceLinkClaim != null ? (String) resourceLinkClaim.get("id") : null;
+        JsonObject resourceLinkClaim = ltiIdToken.getClaim(Claims.RESOURCE_LINK) != null
+                ? JsonParser.parseString(ltiIdToken.getClaim(Claims.RESOURCE_LINK).toString()).getAsJsonObject()
+                : null;
+        this.resourceLinkId = resourceLinkClaim != null ? resourceLinkClaim.get("id").getAsString() : null;
```

Comment:
```suggestion
        var claim = ltiIdToken.getClaim(Claims.RESOURCE_LINK);
        if (claim != null) {
            this.resourceLinkId = JsonParser.parseString(claim.toString()).getAsJsonObject().get("id").getAsString();
        }
        else {
            this.resourceLinkId = null;
        }
```
I feel like this would be easier to read and prevent the usage of ternary operators. What do you think?

Reply:
Thanks for the feedback! I'll adjust the code based on your suggestion for better readability.